app/service/file_svc.py

```diff
@@ -87,6 +87,24 @@ def __init__(self, plugins, exfil_dir):
             return name, xor_file(file_name)
         raise FileNotFoundError
 
+    async def save_file(self, name, content, xored=False):
+        """
+        Save a (payload) file to the stockpile
+        :param name: filename
+        :param content: content to save
+        :param xored: whether or not to xor the contents when saving
+        :return: full path of the saved file
+        """
+        filebase='plugins/stockpile/payloads/'
+        f_content = content
+        filename = str(os.path.join("dummy",name).split(os.path.sep)[-1])
+        if xored:
+            filename = filename + 'xored'
+            f_content = xor_file(content)
+        with open(os.path.join(filebase, filename), 'xb') as file:
```

Comment:
Do we want to fail if the file already exists?

Reply:
Yes (more directory traversal prevention, just in case). If a bad payload is uploaded incorrectly, it should be reasonable to ask the server administrator to remove the existing one, or for users to make a new payload with a different name.